Add networking interface rename command (#1571)
theforeman/foreman-documentation: guides/common/modules/proc_upgrading-project-in-place-using-leapp.adoc

----
endif::[]

> ifdef::satellite[]

Reviewer: Technically, this is not limited to Satellite and can happen on any deployment.

I would however move this down to the `leapp preupgrade` step, and start off with something like:
If `leapp preupgrade` inhibits the upgrade with "Unsupported network configuration" because there are multiple legacy-named network interfaces, please follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names.

Reviewer: For Satellite builds, we might want to link directly to https://access.redhat.com/solutions/4067471, instead of saying "follow the instructions shown by Leapp" ;)

. If there are multiple network interfaces, fix the interface names and rerun the installer with the new interface name.
Call the installer to find out what interface was used and where:
+
----

> # satellite-installer --help |grep 'interface.*eth'

Reviewer 2: Please use attributes instead. You'll also need to allow attributes in the code block.

    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
----

> Because `eth0` was renamed to `em0`, call the installer to rename the interface with:

Reviewer: ```suggestion
If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
```

+
----
# satellite-installer --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
----

> +

Reviewer 2: ```suggestion
```

----
# leapp preupgrade
----

> The first run is expected to fail but report issues.

Reviewer: I think I would keep that here and reword it to something like:
"The first run is expected to report issues and inhibit the upgrade."
And then we can go on and list the possible issues.

Technically, I would probably also merge the next step ("Examine the report") into this one, but probably not in this PR.

> If `leapp preupgrade` inhibits the upgrade with an unsupported network configuration because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names:

Reviewer: "Unsupported network configuration" is the output Leapp gives you -- is there a way to highlight it here as such?
In my original suggestion I quoted it, but not sure this is the correct way to do it?

Author: @evgeni I can make it bold.

----
# leapp preupgrade
----
The first run is expected to fail but report issues and inhibit the upgrade.
+

If `leapp preupgrade` inhibits the upgrade with *Unsupported network configuration* because there are multiple legacy named network interfaces, follow the instructions shown by Leapp to rename the interfaces, followed by an installer run to reconfigure {Project} to use the new interface names:
+
[options="nowrap" subs="attributes"]
----
# {foreman-installer} --help |grep 'interface.*eth'
    --foreman-proxy-dhcp-interface  DHCP listen interface (current: "eth0")
    --foreman-proxy-dns-interface  DNS interface (current: "eth0")
----

> If `eth0` was renamed to `em0`, call the installer to use the new interface name with:

Reviewer 2: ```suggestion

If `eth0` was renamed to `em0`, call the installer to use the new interface name with:
```

+
[options="nowrap" subs="attributes"]
----
# {foreman-installer} --foreman-proxy-dhcp-interface=em0 --foreman-proxy-dns-interface=em0
----
+
Continue to the next step for remediation if the first run fails.

> +

Reviewer 2: ```suggestion
```

. Examine the report in the `/var/log/leapp/leapp-report.txt` file, answer all questions (using `leapp answer`) and manually resolve the other reported problems.
The following commands show the most common steps required:
+

> 

Reviewer 2: ```suggestion
```
